didman/api/v1/api.go

```diff
@@ -86,6 +87,38 @@ func (w *Wrapper) AddEndpoint(ctx echo.Context, didStr string) error {
 	return ctx.NoContent(http.StatusNoContent)
 }
 
+// AddCompoundService handles calls to add a compound service. It only checks params and sets the correct return status code.
+// didman.AddCompoundService does the heavy lifting.
+func (w *Wrapper) AddCompoundService(ctx echo.Context, didStr string) error {
+	request := CompoundServiceCreateRequest{}
+	if err := ctx.Bind(&request); err != nil {
+		err = fmt.Errorf("failed to parse %T: %w", request, err)
+		logging.Log().WithError(err).Warn(problemTitleAddCompoundService)
+		return core.NewProblem(problemTitleAddCompoundService, http.StatusBadRequest, err.Error())
+	}
+
+	id, err := parseDID(didStr, problemTitleAddCompoundService)
+	if err != nil {
+		return err
+	}
+
+	references := make(map[string]ssi.URI, 0)
+	for key, refAsStr := range request.Endpoint {
+		refAsURI, err := ssi.ParseURI(fmt.Sprintf("%s", refAsStr))
```

Comment:
shouldn't the refAsStr be cast to string first (and check the cast)? Otherwise a slice might end up being parsed as URI? Which is probably a mem address but also a valid URI?

Reply:
Yeah you're right, and `nil` will be formatted as "nil", which is a valid URI and thus happily be passed.